csup/primitive.go

```diff
@@ -12,7 +13,7 @@ import (
 
 const MaxDictSize = 256
 
-type PrimitiveEncoder struct {
+type ZcodeEncoder struct {
```

Comment:
If you're going to rename this type, then you should also rename the file and the method receivers. That's a lot of noise unrelated to the goal of this PR so how about doing it in a separate PR?

Reply:
We've taken the name PrimitiveEncoder in this pr so this had to be renamed which I think is fine

Reply:
Call it primitiveEncoder then? (It doesn't actually need to be exported.)

Reply:
I'm all for not exporting this stuff but if you are going to do it there then you need to do it elsewhere. Also I don't like the term `PrimitiveEncoder` because there are already other primitive encoders.